### ISSUE TITLE
DKD: the nckd loss becomes infinity after one iteration of training.

### ISSUE THREAD

**Issue by OuYangg** [megvii-research/mdistiller#60]
First of all, thank the author for open source. DKD is a very good job!

I trained a binary classification model and wanted to use DKD for distillation, but after one iteration of training, the output of the student model became very strange, causing the loss to become Inf.

Some details:
iteration 1:
![image](https://github.com/megvii-research/mdistiller/assets/67104283/8c415dde-4642-45b5-9337-bdd2495a6db2)
![image](https://github.com/megvii-research/mdistiller/assets/67104283/a9380762-f78a-476c-af8b-b293ac5c562b)
iteration 2:
![image](https://github.com/megvii-research/mdistiller/assets/67104283/2e79f539-94ee-406e-9168-c278384e0dcf)

**Comment by OuYangg** [comment]
After I adjusted the learning rate down, the loss is no longer nan, but the nckd loss is still 0.
![image](https://github.com/megvii-research/mdistiller/assets/67104283/dab3e482-cb4e-4108-a92d-38b91d807d69)

**Comment by Zzzzz1** [comment]
NCKD loss can not be applied on binary classification tasks, since there is no "non-target class". The non-target logits will be always 1.0, may introduce some problems when calculating KLD.
